### ISSUE TITLE
bug: using default root logger + log level debug -> infinite recursion

### ISSUE THREAD

**Issue by jossef** [kislyuk/watchtower#65]
what causes this problem is that the `watchtower._idempotent_create()` wraps `botocore._make_api_call()` and there is a `logger.debug` call inside which causes an infinite recursion.

### from `watchtower`
```
   # ...
    def emit(self, message):
        stream_name = self.stream_name
        if stream_name is None:
            stream_name = message.name
        else:
            stream_name = stream_name.format(logger_name=message.name)
        if stream_name not in self.sequence_tokens:
            _idempotent_create(self.cwl_client.create_log_stream,
                               logGroupName=self.log_group, logStreamName=stream_name)
            self.sequence_tokens[stream_name] = None
   # ...
```

### from `botocore.client`
```
    def _make_api_call(self, operation_name, api_params):
        operation_model = self._service_model.operation_model(operation_name)
        service_name = self._service_model.service_name
        history_recorder.record('API_CALL', {
            'service': service_name,
            'operation': operation_name,
            'params': api_params,
        })
        if operation_model.deprecated:
            logger.debug('Warning: %s.%s() is deprecated',
                         service_name, operation_name)
```

My POC script to reproduce this issue looks like this:

```
import logging
import watchtower
import sys

logger = logging.getLogger()

logger_format = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")

cloud_watch_log_handler = watchtower.CloudWatchLogHandler(send_interval=10, log_group='platform', stream_name='monitor')
cloud_watch_log_handler.setFormatter(logger_format)
logger.addHandler(cloud_watch_log_handler)

stream_handler = logging.StreamHandler(sys.stdout)
stream_handler.setFormatter(logger_format)
logger.addHandler(stream_handler)

logger.setLevel(logging.DEBUG)


logger.info('hello world') # <---- maximum recursion depth exceeded while getting the str of an object
```

> maximum recursion depth exceeded while getting the str of an object

**Comment by kislyuk** [comment]
Thank you for reporting. Fix in master, will release asap

**Comment by kislyuk** [comment]
Released in v0.5.4.
